### PR TITLE
Huge Update

### LINKED ISSUE #8: Receive E2E messages
Receiving E2E messages is already supported (at least the processing part). However, an example is missing, yet. A minimal server implementation which can receive and process incoming E2E messages might be interesting, too.

### LINKED ISSUE #15: Improve decryption of E2E messages
The decryption functionality of the Python SDK is insufficient. We should at least be able to decrypt messages we are able to send. Missing types:
- `ImageMessage`
- `FileMessage`

Also required for #8

### LINKED ISSUE #16: Add Script for libsodium installation
Hi,

here is a little script to automate libsodium installation on linux systems.
Need to be executed as root:
[install-sodium.txt](https://github.com/lgrahl/threema-msgapi-sdk-python/files/104709/install-sodium.txt)

### LINKED ISSUE #19: Add Travis CI
Continuous Integration!

### PR DESCRIPTION
**Pull request by lgrahl**
No issues have been filed so far and it has been nearly five months now. Apart from the docs, I don't think there's anything standing against a merge to master.

Furthermore, this is a goodbye to git-flow. We'll use simple GitHub PRs starting from now. The `develop` branch will be removed, `master` will take its place. Any experimental stuff will be done in separate branches. If you need specific releases, please use the corresponding tags.

PRs to merge first:
* [x] #25 
* [x] #26 

Resolves:
* Resolves #19.
* Resolves #16.
* Resolves #15.
* Resolves #8.

### CODE REVIEW

**Comment by lgrahl** [comment]
And idk wtf you're doing Travis, but the config you're loading doesn't even exist. If this isn't fixed after merging, I'll have to google. :-1: 

Edit: Nevermind, I'm an idiot! :grin:

**Comment by lgrahl** [comment]
The tests need some more debugging before this can be merged. I guess it is a problem with the server not being closed properly before new tests start as I had this problem before. I'll add some more logging to the tests and then we'll see what asyncio complains about.

**Comment by lgrahl** [comment]
Tests self-healed. :1st_place_medal:
Edit: No they didn't. But restarting helps. ;D

File: threema/gateway/e2e.py [hunk 1]
```diff
@@ -659,10 +659,15 @@ class DeliveryReceipt(Message):
     """
     @enum.unique
     class ReceiptType(enum.IntEnum):
-        """Describes message receipt types."""
+        """
+        Describes message receipt types.
+
+        .. warning:: `user_ack` is deprecated and will be removed with
+           the next major release. Use `user_acknowledge` instead.
```
Comment (dbrgn) [inline threema/gateway/e2e.py]: We *could* make `user_ack` a property that emits a DeprecationWarning.

Comment (lgrahl) [reply threema/gateway/e2e.py]: Had the same idea but that would have to be a class property and class properties require some fiddling. But feel free to add it.

Comment (dbrgn) [reply threema/gateway/e2e.py]: Ah, then never mind.

An alternative would be to override `__getattribute__` to show the warning, but that's probably overkill :)

**Pull request merged**